contracts/campaignAirdrops/NewERC1155DropCampaign.sol

```diff
@@ -13,17 +13,29 @@ contract NewERC1155DropCampaign is ERC1155Upgradeable, CampaignAidropsShared {
     uint256 public constant tokensPerClaim = 1; // 1 reward per wallet
     string public constant airdropType = "ERC1155";
 
+    string public name;
+    string public symbol;
+    string public contractURI;
     uint256 public airdropFundBlockTimestamp;
 
     address internal airdropFundingHolder;
 
-    function initialize(string memory _uri, address _airbroCampaignFactoryAddress) public initializer {
+    function initialize(
+        string memory _name,
+        string memory _symbol,
+        string memory _uri,
+        address _airbroCampaignFactoryAddress
+    ) public initializer {
         __ERC1155_init(_uri);
         airbroCampaignFactoryAddress = IAirBroFactory(_airbroCampaignFactoryAddress);
```

Comment:
Since this is deployed via factory, you can use msg.sender

Reply:
I agree, great catch @bxmmm1  however :D now that I am not sending that value as a param, in my fixtures.ts:
```
await newSB1155DropCampaign.initialize(name, symbol, uri, mockAirbroCampaignFactory.address); // previous
await newSB1155DropCampaign.initialize(name, symbol, uri); // current
```
how do I mock the value of the address so it's the mocked contract and not actually the msg.sender? Currently it stops me from running my tests, this example above is from my unit tests

Reply:
Especially since it does this:
`airbroCampaignFactoryAddress = IAirBroFactory(msg.sender);
`right away in the initializer @bxmmm1

Reply:
You have multiple ways of doing this.

- Leave as is 😺 
- [Change storage var via hardhat](https://hardhat.org/hardhat-network/docs/reference#hardhat_setstorageat). Deploy mock contract, and change `airbroCampaignFactoryAddress`
- [Deploy with EOA, change bytecode for that EOA so that it becomes a smart contract (mock)](https://hardhat.org/hardhat-network/docs/reference#hardhat_setcode)
- Create `ContractInTest extends NewERC1155DropCampaign` and add a function that changes `airbroCampaignAddress` and use it in tests.

This last one is often used when you want to write unit tests for internal functions of a contract.